### DISCUSSION TITLE
How to update symlinks?

### DISCUSSION THREAD

**Discussion by thorstenwagner** [kevin-hanselman/dud#150] [General]
Hi,

dud ins wonderful. Thank you for that!

One question:

Assume I've a checked out stage with file `set/a1.txt` and `set/a2.txt`

I set a git tag "1and2" for that version.

Now I add `set/a3.txt` and commit+push via `dud`. And also commit&push via `git`.

Next I checkout the "1and2" tag via `git` and want to update the symlinks.

I did `dud fetch` & `dud checkout` but the `a3.txt` file still exists. I had to delete the set folder and do `dud fetch` & `dud checkout ` to get it updated. Probably there is a smarter way?

**Comment by kevin-hanselman** [comment]
Hi @thorstenwagner! Sorry for the late response; I was away on vacation for a while.

You have correctly identified the (currently) recommended way to handle this situation: Delete the old directory/files before checkout. I realize this is not ideal or streamlined, though.

I would like to improve this scenario by adding a `checkout --rm` flag that would delete the relevant artifacts for you prior to the checkout operation. I'm not sure if this would help your situation though, as it seems you are tracking individual files and not a directory. If you're changing a stage's artifacts across commits, Dud does not interrogate Git for past versions of a stage file, so it doesn't know which files were previously tracked or not. However, if you're tracking the entire `set` directory, `checkout --rm` would delete the entire directory before checking it out, thus fixing this issue.

I purposefully did not make `checkout --rm` the default behavior because it adds side effects to the checkout operation. Maintaining simplicity in Dud is a core principle for me.

Thanks again for your question! I'm going to create an issue from this discussion, as I think `checkout --rm` is a sensible feature for Dud.

Issue created: #154

**Reply by thorstenwagner** [reply]
Thanks for reply! Glad that you are back. I was already afraid that dud is not continued :-)
